sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -656,7 +660,7 @@ export class ApplicationService extends RecordDataModelService<Application> {
       },
       where: {
         id: applicationId,
-        applicationStatus: Not(ApplicationStatus.Overwritten),
+        applicationStatus: applicationStatus,
```

Comment:
Minor. condition can be expressed like this. 

![image](https://github.com/user-attachments/assets/93e2eae6-d53c-4af5-8624-13146a565731)